implementation/src/main/java/io/smallrye/jwt/auth/principal/DefaultJWTTokenParser.java

```diff
@@ -86,6 +97,10 @@ public JwtContext parse(final String token, final JWTAuthContextInfo authContext
             JwtClaims claimsSet = jwtContext.getJwtClaims();
 
             claimsSet.setClaim(Claims.raw_token.name(), token);
+            if (authContextInfo.getDefaultSubjectClaim() != null) {
```

Comment:
Please update the condition to start from the `!claimsSet.hasClaim(Claims.sub.name())`

Reply:
Ok.